frontends/bsd/src/screens/admin_actions_screen.test.tsx

```diff
@@ -48,12 +48,17 @@ test('clicking "Export Backup" shows progress', async () => {
     backupButton.click();
     expect(backup).toHaveBeenCalledTimes(1);
 
-    // Verify progress message is shown.
-    await waitFor(() => screen.getByText('Exporting…'));
+    // Verify progress modal is shown.
+    await waitFor(() => {
+      const modal = screen.getByRole('alertdialog');
```

Comment:
Instead of wrapping all three of these lines in a `waitFor`, you could instead rely on `findBy` queries:

```ts
const modal = await screen.findByRole('alertdialog');
```

Reply:
The `isExporting` state comes and goes so quickly, that if you use `findBy` queries, the latter ones will not pass because the component state will have already changed. Wrapping it in a `waitFor` catches a moment where all assertions are true.

Reply:
Ah very interesting, makes sense!